### DISCUSSION TITLE
Make Github Pages domain private.

### DISCUSSION THREAD

**Discussion by devstarttips** [community/community#44593] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Hi team,

I was reading the documentation on Changing the visibility of your Github Pages Site: https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site

My question is, is there a way to make the deployment link created in Github Pages only visible to a certain group of users. The guide mentions making the repository Private which I understand, and also mentions a private sub-domain, but I'm unclear as to whether this would limit visibility of the deployed link to certain users.

Would there be a way, after making the repository Private, to have a domain link that would only be visible to certain users?

Thanks!

**Comment by yoannchaudet** [comment]
Hello,

Private visibility for GitHub Pages is only available to private repositories. As such and because we only gate Pages access to whoever has read access on the repository, the persons who can see the deployments have access to the site already.

**Comment by nvspavankalyanch** [comment, marked as answer]
Yes, there is a way to make the deployment link created in GitHub Pages only visible to a certain group of users.
One way to do this is to set up a private subdomain for your GitHub Pages site. This would allow you to share the link to the site with a specific group of people while keeping the repository private. Additionally, you can also use GitHub's built-in access control features, such as Collaborators, to limit access to the repository and its pages to specific users or teams. You can also use third party apps such as 'Probot App' that can restrict access to the repository and its pages to specific users or teams.                                                                                                             
                                                                                                                                                                                                                    

Here is a step-by-step process for setting up a private subdomain for your GitHub Pages site and limiting access to specific users using **Probot App**:

- Create a new repository for your GitHub Pages site and make it private.

- Set up a custom subdomain for your GitHub Pages site by following the instructions in the GitHub Pages documentation.

- Install the Probot App by visiting the GitHub marketplace and searching for **Probot App** then install it.

- Go to the repository's settings, find the "Integrations & services" section, and add **Probot App** to the repository.

- Configure 'Probot App' by following the instructions provided by the app, this will allow you to restrict access to the repository and its pages to specific users or teams based on their GitHub usernames, teams, or organizations.

- Share the link to the private subdomain with the specific users or teams that you have granted access to. They will be able to access the site using that link, while others will not be able to access it.

By following these steps, you will be able to set up a private subdomain for your GitHub Pages site and limit access to specific users or teams using **Probot App**

**Reply by devstarttips** [reply]
Thank you much, I'll give this a try!
